### ISSUE TITLE
Uplevel winpty to 0.4.1

### ISSUE THREAD

**Issue by Tyriar** [Tyriar/node-pty#10]
[This release](https://github.com/rprichard/winpty/releases/tag/0.4.0) seems to have fixed the arrow key in WSL issue https://github.com/Microsoft/vscode/issues/10163.

**Comment by stereokai** [comment]
Good luck! Let me know if you need any help.

**Comment by whitecolor** [comment]
Is it goint to be released soon?

**Comment by Tyriar** [comment]
Potentially as part of vscode 1.9, I was blocked on this due to bug in 0.4.0 that was fixed in master but not released, it got released after I reached out to the author a couple of days ago https://github.com/rprichard/winpty/releases

It is a stretch goal though as the other issues I'm working on are higher priority https://github.com/Microsoft/vscode/issues/17608

**Comment by Tyriar** [comment]
Success! https://github.com/Tyriar/node-pty/pull/20

**Comment by Tyriar** [comment]
Running into some issues introduced by winpty@0.4.1 which seem to make some things worse. FWIW bash on windows seems really solid, provided it's running `tmux` 😕

**Comment by whitecolor** [comment]
@Tyriar 
> FWIW bash on windows seems really solid, provided it's running tmux 

I actually has many troubles now with vscode terminal (after multiple commands the whole command line input becomes corrupted - different symbols appear and disappear when try to edit the command) and especially with `tmux` the are many visual bugs. I posted screens today in some issue related to tmux.

**Comment by Tyriar** [comment]
Yeah, unfortunately there seem to be even more issues when running on winpty@0.4.1 unless I'm missing something; things such as:

- Bash on Windows: arrows working, but not in vim
- Git Bash: backspace moving cursor in opposite direction.

Bash on windows with tmux on winpty@0.4.1 seems really good for some reason though.
